src/Workspaces/Core/Portable/Differencing/LongestCommonSubsequence.cs

```diff
@@ -190,18 +190,11 @@ public IEnumerable<(VArray Array, int Depth)> ConsumeArrays()
         }
 
         // VArray struct enables array indexing in range [-d...d].
-        protected readonly struct VArray
+        protected readonly struct VArray(int[] buffer, int start, int length)
         {
-            private readonly int[] _buffer;
-            private readonly int _start;
-            private readonly int _length;
-
-            public VArray(int[] buffer, int start, int length)
-            {
-                _buffer = buffer;
-                _start = start;
-                _length = length;
-            }
+            private readonly int[] _buffer = buffer;
```

Comment:
>buffer [](http://example.com/codeflow?start=45&length=6)

I don't think I'd keep the fields and also use primary ctor at the same time. Either, or. As it is now, both `buffer` and `_buffer` are in scope in all methods. If someone accidentally uses `buffer` instead of `_buffer` there is no error but the struct becomes larger.

Reply:
Compiler will warn in that case. Assigning to field, but also using the parameter will warn about double storage :-)

Reply:
I don't see the warning in Version 17.7.0 Preview 2.0 [33722.583.main]
![image](https://github.com/dotnet/roslyn/assets/41759/9a6f28cf-0f8a-43a3-9aa9-0e701f652083)
Is it new?

Reply:
Tracked here: https://github.com/dotnet/csharplang/blob/main/meetings/2023/LDM-2023-05-15.md#primary-constructors

LDM just approved it.  Will be coming asap.